app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -114,13 +131,31 @@ def collect_legacy_inventory_for_targets(ems)
         [[ems, nil]]
       end
 
-      def parse_targeted_inventory(ems, target, inventory)
+      def parse_targeted_inventory(ems, target, collector)
         # legacy refreshers collect inventory during the parse phase
         # new refreshers should override this method to parse inventory
         # TODO: remove this call after all refreshers support retrieving
         # inventory separate from parsing
-        hashes, _ = Benchmark.realtime_block(:parse_legacy_inventory) { parse_legacy_inventory(ems) }
-        hashes
+        if collector.kind_of? ManagerRefresh::Inventory::Collector
+          log_header = format_ems_for_logging(ems)
+          _log.debug "#{log_header} Parsing inventory..."
+          inventory_collections, = Benchmark.realtime_block(:parse_inventory) do
+            provider_module = ManageIQ::Providers::Inflector.provider_module(ems.class).name
+            inventory_target_class = "#{provider_module}::Inventory::Targets::#{target.class.name.demodulize}".safe_constantize
+            inventory_target = inventory_target_class.new(target)
+
+            parser_class = "#{provider_module}::Inventory::Parsers::#{target.class.name.demodulize}".safe_constantize
+            parser = parser_class.new
+
+            i = ManagerRefresh::Inventory.new(inventory_target, collector, parser)
+            i.inventory_collections
+          end
+          _log.debug "#{log_header} Parsing inventory...Complete"
+          inventory_collections
+        else
+          hashes, = Benchmark.realtime_block(:parse_legacy_inventory) { parse_legacy_inventory(ems) }
```

Comment:
Ah, right ... this is where I was really thinking should be inlined.  But, this is truly delegated out to the providers for the legacy inventory collection.  I guess it can stay for now, but I'm looking forward to this dying off soon.